frost-ristretto255/src/frost.rs

```diff
@@ -576,43 +856,37 @@ fn generate_lagrange_coeff(
 /// the commitment that was assigned by the coordinator in the SigningPackage.
 pub fn sign(
     signing_package: &SigningPackage,
-    participant_nonces: SigningNonces,
-    share_package: &SharePackage,
+    signer_nonces: &SigningNonces,
+    signer_commitments: &SigningCommitments,
+    key_package: &KeyPackage,
 ) -> Result<SignatureShare, &'static str> {
-    // TODO(dconnolly): tidy up now that rho is the same for all i's
-
-    let mut bindings: HashMap<u64, Scalar> =
-        HashMap::with_capacity(signing_package.signing_commitments.len());
-
-    let rho = generate_rho(signing_package);
+    let rho: Rho = signing_package.into();
 
-    for comm in signing_package.signing_commitments.iter() {
-        bindings.insert(comm.index, rho);
-    }
-
-    let lambda_i = generate_lagrange_coeff(share_package.index, signing_package)?;
-
-    let group_commitment = generate_group_commitment(signing_package, &bindings)?;
+    let group_commitment = GroupCommitment::try_from(signing_package)?;
```

Comment:
Similarly, this line needs to be changed to use values strictly sent from round one.